### PR TITLE
static_map.h: fix lowercase convert

### PR DESCRIPTION
**Pull request by 2020karaulov2020**
**REHLDS_OPT_PEDANTIC** static_map.h fix

`if (true) {	cc |= 0x20; } `
to 
`if ( cc >= 'A' && cc <= 'Z') { cc |= 0x20;	}`

### CODE REVIEW

**Comment by 2020karaulov2020** [comment]
NEED ?

**Comment by RauliTop** [comment]
seems to be a typo

File: rehlds/public/rehlds/static_map.h [hunk 1]
```diff
@@ -229,8 +229,8 @@ class CICaseStringKeyStaticMap : public CStaticMap<const char*, T_VAL, ASSOC_2N,
 		if (cpuinfo.sse4_2) {
 			while (*pcc) {
 				char cc = *(pcc++);
-				if (cc >= 'A' || cc <= 'Z') {
-					cc |= 0x20;
+				if (cc >= 'A' && cc <= 'Z') {
+					cc += ('a' - 'A');
```
Comment (WPMGPRoSToTeMa) [inline rehlds/public/rehlds/static_map.h:233]: IMO that is more readable:
```suggestion
					cc = (cc - 'A') + 'a';
```

Comment (wopox1337) [reply rehlds/public/rehlds/static_map.h:233]: I would like to use Q_strlwr but....

**Review by wopox1337 — APPROVED** [review]

File: rehlds/public/rehlds/static_map.h (hunk 1 above)
Comment (wopox1337) [reply rehlds/public/rehlds/static_map.h]: I would like to use tolower() but....

Thanks. Done b7809bd

**Pull request merged**